### ISSUE TITLE
Route model binding broken in 11.36.0

### ISSUE THREAD

**Issue by jasonvarga** [laravel/framework#53952]
### Laravel Version

11.36.0

### PHP Version

8.3.14

### Description

Explicit route model binding is broken in 11.36.0, specifically in https://github.com/laravel/framework/pull/53275 in [this condition and early return](https://github.com/laravel/framework/pull/53275/files#diff-453d4a1b4e60e4a93d33f4845b551ddfd3d2d7fa0c1ab363c1f29d1adee38f6fR98-R102)

When I have a route like this:

```php
Route::get('things/{thing}', function (ThingInterface $thing) {
    dump($thing);
});
```

and a binding like this:

```php
Route::bind('thing', function (string $value) {
    return new Thing($value);
});
```

The route parameter will try to resolve from the container rather than use what my route binding returns.

### Steps To Reproduce

I have an example at https://github.com/jasonvarga/laravel-bug-pr-53275
[This commit](https://github.com/jasonvarga/laravel-bug-pr-53275/commit/33628ac7a761ff5b9fa2e1fee913690bb02ef9a8#diff-41d6d45248b6a715d8d16d40ea319bca836d38c4f5f2d19cd477e5e9eaafbd94R5) shows the proof of concept.

Visit `/things/foo` to see the issue. You will see a BindingResolutionException because it ignores the binding and tries to resolve through the container.

If you downgrade to the previous release (11.35.1) then you will see the returned `Thing` correctly dumped to the browser.

**Comment by crynobone** [comment]
Fixed in v11.36.1: https://github.com/laravel/framework/releases/tag/v11.36.1

**Comment by jasonvarga** [comment]
Thank you
